cmd/deploy/deploy.go

```diff
@@ -362,3 +364,29 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 	return handler, nil
 
 }
+
+func addEnvVars(ctx context.Context, cwd string) error {
```

Comment:
what's this for?

Reply:
I'm adding this because we were doing it on the backend and it's documented [here](https://okteto.com/docs/cloud/okteto-pipeline/#built-in-environment-variables)

Reply:
hmm, I think that populating `OKTETO_GIT_COMMIT` env var for `okteto deploy` could cause some issues with the build optimization.

Let say, I have the local repo movies cloned with some local changes I want to test. If I run it and it tries to build the image `okteto.dev/api:${OKTETO_GIT_COMMIT}` but there is already an image built with that commit which doesn't contain my local changes, I wouldn't be deploying exactly what I want to deploy because it would reuse the one already built.

Reply:
@jLopezbarb what value is returned by `utils.GetGitCommit` if you have uncommitted changes?

Reply:
The value is the same, is the last commit until you create a new one. One solution could be:
If there are no new changes get the commit.
If there are local changes add to the commit sha the suffix local.
This option will fix the first time of deploying it but if you want to redeploy it will deploy the changes that you did in the first deploy commit. @ifbyol suggested me to inject a suffix or special tag that skips the build optimisation.
Another solution could be show a warning to the user telling that the `OKTETO_GIT_COMMIT` env var has not been injected due to having local changes 
What do you think could be the best solution?

Reply:
@jLopezbarb I would add a random suffix, or we can also ignore the build optimization if `OKTETO_GIT_COMMIT` ends with `local`. Both of them work.